forwardproxy.go

```diff
@@ -45,6 +45,8 @@ type ForwardProxy struct {
 	dialTimeout        time.Duration // for initial tcp connection
 	hostname           string        // do not intercept requests to the hostname (except for hidden link)
 	port               string        // port on which chain with forwardproxy is listening on
+	upstream           string
+	dial               func(network, address string) (net.Conn, error)
```

Comment:
`upstream` store the upstream server addr 
`dial` is a dial function for upstream proxy

Reply:
Those are good comments, and they belong in code for future readers to see.